ui/src/tasks/components/TaskRunsPage.tsx

```diff
@@ -1,25 +1,47 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
 
 // Types
-import {Run} from '@influxdata/influx'
 import {Page} from 'src/pageLayout'
 import TaskRunsList from 'src/tasks/components/TaskRunsList'
+import {AppState} from 'src/types/v2'
 
-// DummyData
-import {taskRuns} from 'src/tasks/dummyData'
+// Actions
+import {getRuns} from 'src/tasks/actions/v2'
+import {Run} from '@influxdata/influx'
+import {RemoteDataState} from 'src/types'
 
-interface Props {
+interface PassedInProps {
   params: {id: string}
+}
+
+interface ConnectedDispatchProps {
+  getRuns: typeof getRuns
+}
+
+interface ConnectedStateProps {
   runs: Run[]
 }
 
-const dummyData = taskRuns
+interface State {
+  loading: RemoteDataState
```

Comment:
It seems like storing the runs in redux but storing the loading state for the runs outside of redux could potentially cause us problems in the future. In particular,

- It splits state across the application, which makes thinking about state more difficult (and thinking about state is the hardest part of building a web app)
- The state can get out of sync if the `TaskRunsPage` is unmounted and remounted for some reason. A react component should be able to be unmounted/remounted/rerendered whenever and still accurately reflect the state of the application

I know we do this pattern elsewhere a lot, but it's something I would like to change.

What do you think about adding a `runsStatus: RemoteDataState` property to the task reducer and storing the state there instead?

Then the `setRuns` action could look something like this instead:

```
export interface SetRuns {
  type: 'SET_RUNS'
  payload: {
    runs: Run[]
    runsStatus: RemoteDataState
  }
}
```

Reply:
Yes, that makes more sense! Thanks for reviewing :)